pkgs/tools/security/hashdeep/0000-fix-cpp-defect-report-1512.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/hash.cpp b/src/hash.cpp
+index 4216157..da85bb5 100644
+--- a/src/hash.cpp
++++ b/src/hash.cpp
+@@ -279,7 +279,7 @@ void file_data_hasher_t::hash()
+ 		MAP_FILE|
+ #endif
+ 		MAP_SHARED,fd,0);
+-	    if(fdht->base>0){		
++	    if(fdht->base){
```

Comment:
This isn't semantically the same unless `base` is unsigned.

Reply:
You are right, for some reason when `mmap` fails, it doesn't return `NULL` but `(void*) -1`. I corrected the check to verify if it is different than `MAP_FAILED`, i.e. what the could should actually do.